test/unit/protip/wrapper_test.rb

```diff
@@ -414,6 +462,20 @@ module Protip::WrapperTest # namespace for internal constants
         assert_equal inner_message_class.new(value: 50), wrapper.message.inner
       end
 
+      it 'for nested resources, sets the resource\'s message' do
+        message = wrapped_message
+        klass = resource_class
+        new_inner_message = inner_message_class.new(value: 50)
+
+        resource = klass.new new_inner_message
+        wrapper = Protip::Wrapper.new(message, Protip::StandardConverter.new, {inner: klass})
+
+        resource.expects(:message).once.returns(new_inner_message)
+        wrapper.inner = resource
+
+        assert_equal new_inner_message, wrapper.message.inner
```

Comment:
overall, it's hard to follow this test IMO. one thing that might help would be to add an error message as the third arg describing the failure case.